go/cli/mcap/cmd/merge.go

```diff
@@ -367,12 +368,124 @@ func (m *mcapMerger) mergeInputs(w io.Writer, inputs []namedReader) error {
 		}
 		heap.Push(pq, utils.NewTaggedMessage(msg.InputID, newMessage))
 	}
+
+	// append any attachments as they are encountered. if an input is unindexed,
+	// we do a second full scan of it.
+	for _, input := range inputs {
+		_, err := input.reader.Seek(0, io.SeekStart)
+		if err != nil {
+			return fmt.Errorf("failed to seek: %w", err)
+		}
+
+		reader, err := mcap.NewReader(input.reader)
+		if err != nil {
+			return fmt.Errorf("failed to create attachment reader: %w", err)
+		}
+		defer reader.Close() //nolint:gocritic // we actually want these deferred in the loop.
+		info, err := reader.Info()
+		if err == nil && info != nil && info.Statistics != nil {
+			if info.Statistics.AttachmentCount > 0 {
```

Comment:
we should be checking for `info.Statistics.AttachmentCount != 0 && info.Statistics.AttachmentCount == len(info.AttachmentIndexes)`. This accounts for mcaps that have a statistics record but no attachment indexes.